### PR TITLE
Add routes for templates as json

### PR DESCRIPTION
**Pull request by PhilipGough**
ping @philbrookes

### CODE REVIEW

**Comment by PhilipGough** [comment]
### Verification steps

1. Run and hit the /service/templates route and ensure the two templates are dumped as valid json

File: web/routes.go [hunk 1]
```diff
@@ -56,3 +56,9 @@ func SysRoute(r *mux.Router) {
 func Metrics(r *mux.Router) {
 	r.Handle("/metrics", prometheus.Handler()).Methods("GET")
 }
+
+// Service template route
```
Comment (philbrookes) [inline web/routes.go]: Go comments are suggested to start with the function name they are commenting, like:
```go
// Templates sets up the template route handler for the web interface.
```

File: web/template.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+	"fmt"
+
+	"github.com/feedhenry/negotiator/deploy"
+	"github.com/feedhenry/negotiator/pkg/openshift"
+)
+
+// TemplateHandler is the handler for the /services/template route
+type TemplateHandler struct {
+	tl deploy.TemplateLoader
```
Comment (philbrookes) [inline web/template.go]: would it be worth using more verbose names here, so the later code is easier to decipher? 
i.e.
```go
    templateLoader deploy.TemplateLoad
    templateDecoder deploy.TemplateDecoder
```

File: web/template.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	return TemplateHandler{
+		tl: tl, td: td,
+	}
+}
+
+// List all available templates
+func (s TemplateHandler) List(res http.ResponseWriter, req *http.Request) {
+	res.Header().Add("Content-type", "application/json")
+	var err error
+	var t *template.Template
```
Comment (philbrookes) [inline web/template.go]: as above, I think:
```go
var template *template.Template
var templates []*deploy.Template
```

 Would be easier to decipher into the future?

File: web/template.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		var buf bytes.Buffer
+		if err := t.ExecuteTemplate(&buf, v, &deploy.Payload{}); err != nil {
+			fmt.Println(err)
+		}
+
+		decoded, err := s.td.Decode(buf.Bytes())
+		if err != nil {
+			fmt.Println(err)
+		}
+		ts = append(ts, decoded)
```
Comment (philbrookes) [inline web/template.go]: I might have misread this code, but it looks like this is returning the entire template in the JSON response?
Would we not want the response to be something like:
```json
{
    "services": ["cache", "mongodb"]
}
```

File: web/template.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	var err error
+	var t *template.Template
+	var ts []*deploy.Template
+	for v := range openshift.PackagedTemplates {
+		t, err = s.tl.Load(v)
+		if err != nil {
+			fmt.Println(err)
+		}
+		var buf bytes.Buffer
+		if err := t.ExecuteTemplate(&buf, v, &deploy.Payload{}); err != nil {
```
Comment (philbrookes) [inline web/template.go]: I believe this payload would be empty, as this request doesn't have any values in it for populating the templates?

Comment (philbrookes) [inline web/template.go]: oh I see this is actually just an empty payload, I had misread this.

**Pull request merged**